pickLayer/plugin.py

```diff
@@ -52,11 +54,97 @@ def __init__(self) -> None:
             pass
 
         self.actions: List[QAction] = []
+        self.toolbar: Optional[QToolBar] = None
         self.menu = "&pick layer"
-        self.pick_layer: Optional[PickLayer] = None
+        self.pick_layer_tool: Optional[PickLayer] = None
         self.pick_layer_action: Optional[QAction] = None
+        self.set_active_layer_tool = SetActiveLayerTool(iface.mapCanvas())
+        self.set_active_layer_action: Optional[QAction] = None
 
-    def add_action(
+    def get_set_active_layer_tool_action(self) -> QAction:
+        """
+        Public method for getting action that sets active layer.
+
+        Can be used to set shortcut key from other plugins.
+        """
+        return self.set_active_layer_action
+
+    def set_active_layer_using_closest_feature(
+        self, point_xy: QgsPointXY, search_radius: Optional[float] = None
+    ) -> None:
+        """
+        Public method for setting layer active based on given map coordinates.
+
+        If multiple features are found, active layer is selected using this order:
+        vectorlayer(point), vectorlayer(linestring), vectorlayer(polygon),
+        other layers.
+
+        Args:
+            point_xy: Map coordinates
+            search_radius: Search radius to use in map units. By default uses
+              search radius defined in PickLayer settings.
+        """
+
+        canvas_point = QgsMapTool(iface.mapCanvas()).toCanvasCoordinates(point_xy)
+
+        self.set_active_layer_tool.set_active_layer_using_closest_feature(
+            canvas_point.x(), canvas_point.y(), search_radius
```

Comment:
koska on olemassa identify-metodi, joka hyväksyy geometrioita (https://github.com/qgis/QGIS/blob/master/src/gui/qgsmaptoolidentify.h#L139), ja identify toolin omakin implementaatio vaan kääntää canvas x-y:t geometryksi ja käyttää sitä samaa (https://github.com/qgis/QGIS/blob/master/src/gui/qgsmaptoolidentify.cpp#L108), oisko tää public api helpointa toteuttaa myös niin et toi oiski karttatyökalulla `set_active_layer_using_closest_feature(geometry: QgsGeometry, ...)`, ja sit se mouseClick kääntäis canvas x-y:t geometrytksi ja kutsuis tota geometryllä? sithän tässä ei tartte kääntää koordinaatteja canvaksen kautta missään vaiheessa.

Reply:
good point, made the suggested change